tests/test_iqgen_main.py

```diff
@@ -0,0 +1,946 @@
+  params = [
+      '--gps-sv', '1',
+      '--bands', 'l1ca',
+      '--ignore-code-doppler', '1'
+  ]
+  args = parser.parse_args(params)
+  doppler = args.gps_sv[0].getDoppler()
+  assert doppler.isCodeDopplerIgnored() == True
+
+
+def test_parameters_TCXO0():
+  '''
+  TCXO parameter test
+  '''
+  parser = prepareArgsParser()
+  params = [
+      '--tcxo-type', 'sine',
+      '--tcxo-a0', '0',
+      '--tcxo-a1', '1',
+      '--tcxo-period', '3']
+  args = parser.parse_args(params)
+  assert isinstance(args.tcxo, TCXOSine)
+
+
+def test_parameters_TCXO1():
+  '''
+  TCXO parameter test
+  '''
+  parser = prepareArgsParser()
+  params = [
+      '--tcxo-type', 'poly',
+      '--tcxo-a1', '1']
+  args = parser.parse_args(params)
+  assert isinstance(args.tcxo, TCXOPoly)
+
+
+def test_parameters_TCXO2():
+  '''
+  TCXO parameter error test
+  '''
+  parser = prepareArgsParser()
+  params = [
+      '--tcxo-type', 'abc',
+      '--tcxo-a0', '0',
+      '--tcxo-a1', '1']
+  try:
+    parser.parse_args(params)
+    assert False
+  except:
+    pass
+
+
+def test_parameters_msgtype0():
+  '''
+  All Zero message test
+  '''
+  parser = prepareArgsParser()
+  params = [
+      '--gps-sv', '1',
+      '--message-type', 'zero']
+  args = parser.parse_args(params)
+  assert isinstance(args.gps_sv[0].getL1CAMessage(), ConstMessage)
+  assert args.gps_sv[0].getL1CAMessage().bitValue == 1
+
+
+def test_parameters_msgtype1():
+  '''
+  All One message test
+  '''
+  parser = prepareArgsParser()
+  params = [
+      '--gps-sv', '1',
+      '--message-type', 'one']
+  args = parser.parse_args(params)
+  assert isinstance(args.gps_sv[0].getL1CAMessage(), ConstMessage)
+  assert args.gps_sv[0].getL1CAMessage().bitValue == -1
+
+
+def test_parameters_msgtype2():
+  '''
+  Zero+One message test
+  '''
+  parser = prepareArgsParser()
+  params = [
+      '--gps-sv', '1',
+      '--message-type', 'zero+one']
+  args = parser.parse_args(params)
+  assert isinstance(args.gps_sv[0].getL1CAMessage(), ZeroOneMessage)
+
+
+def test_parameters_msgtype3():
+  '''
+  GPS CRC messages test
+  '''
+  parser = prepareArgsParser()
+  params = [
+      '--gps-sv', '1',
+      '--message-type', 'crc']
+  args = parser.parse_args(params)
+  assert isinstance(args.gps_sv[0].getL1CAMessage(), LNavMessage)
+  assert isinstance(args.gps_sv[0].getL2CMessage(), CNavMessage)
+
+
+def test_parameters_msgtype4():
+  '''
+  GLONASS CRC message test
+  '''
+  parser = prepareArgsParser()
+  params = [
+      '--glo-sv', '0',
+      '--message-type', 'crc']
+  args = parser.parse_args(params)
+  assert isinstance(args.gps_sv[0].getL1Message(), GLOMessage)
+
+
+def test_printOutput():
+  '''
+  Plain configuration output test
+  '''
+  parser = prepareArgsParser()
+  params = [
+      '--gps-sv', '1',
+      '--bands', 'l1ca+l2c',
+      '--glo-sv', '1',
+      '--bands', 'l1+l2',
+  ]
+  args = parser.parse_args(params)
+  printOutputConfig(NormalRateConfig, args)
```

Comment:
Let's add a comment that printOutputConfig() should not generate an exception to pass the test.